src/main/java/hmim/eteam/rest/backend/controller/CourseController.java

```diff
@@ -0,0 +1,155 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class CourseController {
+    private final IRoleResolver roleResolver;
+    private final CourseRepository courseRepository;
+    private final CourseRoleRepository courseRoleRepository;
+
+    // Adhook for ::courses.
+    private final ThemeStatusRepository themeStatusRepository;
+    private final CourseThemeRepository courseThemeRepository;
+    private final SiteUserRepository siteUserRepository;
+
+    public CourseController(IRoleResolver roleResolver, CourseRepository courseRepository,
+                            CourseRoleRepository courseRoleRepository,
+                            ThemeStatusRepository themeStatusRepository,
+                            SiteUserRepository siteUserRepository,
+                            CourseThemeRepository courseThemeRepository) {
+        this.roleResolver = roleResolver;
+        this.courseRepository = courseRepository;
+        this.courseRoleRepository = courseRoleRepository;
+        this.themeStatusRepository = themeStatusRepository;
+        this.siteUserRepository = siteUserRepository;
+        this.courseThemeRepository = courseThemeRepository;
+    }
+
+    public ResponseEntity<List<CourseRole>> courseRoles(String token, @NotNull Long courseId) {
+        if (token == null) {
+            return new ResponseEntity<>(HttpStatus.FORBIDDEN);
+        }
+
+        Optional<hmim.eteam.rest.backend.entity.course.Course> course = courseRepository.findById(courseId);
+        if (!course.isPresent()) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+
+        UserRole role = roleResolver.resolve(token, courseId);
+        if (role == UserRole.Admin) {
+            List<hmim.eteam.rest.backend.entity.user.CourseRole> allRoles =
+                    courseRoleRepository.findByCourse(course.get());
+
+            List<CourseRole> allRolesRepresentation = new ArrayList<>();
+            allRoles.forEach(foundRole -> allRolesRepresentation.add(foundRole.toApiRepresentation()));
+            return new ResponseEntity<>(allRolesRepresentation, HttpStatus.OK);
+
+        } else {
+            return new ResponseEntity<>(HttpStatus.FORBIDDEN);
+        }
+    }
+
+    public ResponseEntity<List<Course>> courses(String token) {
+        List<Course> courses = new ArrayList<>();
+
+        courseRepository.findAll().forEach(sourceCourse ->
+        {
+            // Adhook, but ok for now.
+            Course course = new Course();
+            course.id(sourceCourse.getId()).
+                    name(sourceCourse.getName()).
+                    priority(sourceCourse.getPriority())
+                    .participantsQuantity(
+                            courseRoleRepository.countDistinctByCourseAndRole(sourceCourse, UserRole.Student) +
+                                    courseRoleRepository.countDistinctByCourseAndRole(sourceCourse, UserRole.Admin));
```

Comment:
this makes thesame table scanned twice. i think, writing a custom query here is an important optimization:
`select count(distinct ...) + count(distinct ...) from ...`